### ISSUE TITLE
Missing fieldMetadata content for multiple relation widget in preview

### ISSUE THREAD

**Issue by StevenDepa** [netlify/netlify-cms#4177]
I cannot show the content of a multiple relation widget inside a custom preview, since the fieldMetadata prop doesn't contain the data anymore.

**To Reproduce**
1. Create a relation field on a collection type with multiple true, referencing another collection type (es. post authors)
2. Populate the related entry (es. create some authors)
3. Populate the relation widget referencing the entry (es. insert in blog post related authors)
4. Register a custom preview component for the entry with the relation field

**Expected behavior**
The props passed to the preview component should contain the data for the relation entities, nested inside the fieldMetadata prop. This should allow the preview component to display the entry like in production. The expected content is like in https://github.com/netlify/netlify-cms/issues/2434#issuecomment-510563966.

**Applicable Versions:**
 - netlify-cms-app version: 2.14.19
 - Git provider: GitHub
 - OS: Windows 8
 - Browser version: latest Chrome
 - Node.JS version: 12.16.1

**Additional context**
The bug doesn't appear in netlify-cms-app 2.14.18. By exclusion, in my investigation I think it was probably introduced in #4063.

**Comment by barthc** [comment]
@StevenDepa trying to confirm your issue, the `fieldMetaData` is passed down to the preview. Can you show some code from your end?

**Comment by StevenDepa** [comment]
It'll be a little complicated to share the entire setup, so I put together a way to reproduce it:
clone the starter [gatsby-starter-netlify-cms](https://github.com/netlify-templates/gatsby-starter-netlify-cms), then install:
netlify-cms-app@2.12.19
gatsby-plugin-netlify-cms@4.3.11
netlify-cms-proxy-server@1.3.1

My website is not live yet, so I'm working on local_backend, so I installed **netlify-cms-proxy-server** and added:
```
const { registerLocalFs } = require('netlify-cms-proxy-server/dist/middlewares');
exports.onCreateDevServer = async ({ app }) => {
  console.log('start local proxy server')
  await registerLocalFs(app);
};
```
to gatsby-node.js, to avoid running another process.
Then in config.yml I added:
```
local_backend: 
  url: http://localhost:8000/api/v1
```
to the top.
Then I added:
```
- {label: "Related posts", name: "related", widget: "relation", 
          collection: "blog", multiple: true, valueField: "{{slug}}", searchFields: ["title"], displayFields: ["title"]}
```
to the _blog_ collection config, to have a relation field.
Then in BlogPostPreview:
```
const BlogPostPreview = ({ entry, widgetFor, fieldsMetaData }) => {
  console.log('fieldsMetaData: ', fieldsMetaData);
```
The run **gatsby develop**, go to localhost:8000/admin, add some related post to one, and open the console, you'll find that the _fieldsMetaData_ field is almost empty.
Finally, if you try to install netlify-cms-app@2.12.18, and repeat, you'll find that the prop is populated correctly.
Specifically, you'll find that the **_root** property is populated.

**Comment by barthc** [comment]
@StevenDepa  thanks for the detailed setup. However, I was still able to access the `fieldsMetaData`  with your setup. On select of relation item, the `fieldsMetaData` is available in-store and is passed to the preview.

**Comment by StevenDepa** [comment]
Hmmm, maybe I wasn't clear: the field is always present, but in the recent version it's almost empty, I cannot find in it the data about the related items.

**Comment by barthc** [comment]
> Hmmm, maybe I wasn't clear: the field is always present, but in the recent version it's almost empty, I cannot find in it the data about the related items.

Yes, the data of the selected items is present from my end.

**Comment by StevenDepa** [comment]
I found the problem: when you select a relation item, the fieldsMetaData content shows up, but not on first load. So when someone enters to see an already filled entry it does not see the content in the preview.

**Comment by StevenDepa** [comment]
This is still a bug in my opinion. I should be able to see a full preview of a filled entry, without having to select something.

**Comment by StevenDepa** [comment]
I've seen you have released the fix only yesterday. I tested the update and it doesn't work. The _fieldsMetadata_ value is not empty anymore, but the content goes down only to the related field id, at the end there's no content, so it's useless.

**Comment by barthc** [comment]
> but the content goes down only to the related field id, at the end there's no content,

Not sure I understand the point here. In your case, can you show the console log of the value of your `fieldsMetaData`.

**Comment by StevenDepa** [comment]
My content has a field (multiple) called **events**, which are relation for a collection called **event**, and in this case the event slug is called"**parental-control**, which is the valueField. I'm not able to find other content of the related object, like before.


![fieldsmetadata](https://user-images.githubusercontent.com/11633401/93241464-a76ef800-f785-11ea-8450-b8fc19251812.png)

**Comment by djfrsn** [comment]
I've been experiencing the same issue and am using the latest version 2.12.22. I've tested this today after watching this issue progress. The below screenshot shows the values being logged from fieldsMetaData in a custom preview template. The fields being used are two relation widgets with multiple set true. 

![fieldsMetaDataExample](https://user-images.githubusercontent.com/2959169/93243395-fd6b7d80-f73c-11ea-9837-be6bdd27b0f1.png)

**Comment by erezrokah** [comment]
Re-opening pending further investigation

**Comment by barthc** [comment]
I was able to reproduce the issue using string template as `value_field`. I opened another PR for a fix here  #4326
